src/lib.ts

```diff
@@ -46,58 +46,75 @@ function clamp( val: number | string, min: number, max: number ): number {
 /**
  * Get a file from S3/
  */
-export async function getS3File( config: Config, key: string, args: Args ): Promise<GetObjectCommandOutput> {
-	const s3 = new S3Client( {
-		...config,
-		signer: {
-			/**
-			 *
-			 * @param request
-			 */
-			sign: async request => {
-				if ( ! args['X-Amz-Algorithm'] ) {
-					return request;
-				}
-				const presignedParamNames = [
-					'X-Amz-Algorithm',
-					'X-Amz-Content-Sha256',
-					'X-Amz-Credential',
-					'X-Amz-SignedHeaders',
-					'X-Amz-Expires',
-					'X-Amz-Signature',
-					'X-Amz-Date',
-					'X-Amz-Security-Token',
-				] as const;
-				const presignedParams: { [K in ( typeof presignedParamNames )[number]]?: string } = {}; // eslint-disable-line no-unused-vars
-				const signedHeaders = ( args['X-Amz-SignedHeaders']?.split( ';' ) || [] ).map( header => header.toLowerCase().trim() );
+export async function getS3File(config: Config, key: string, args: Args): Promise<GetObjectCommandOutput> {
+    const isPresigned = !!args['X-Amz-Algorithm'];
+    const unauthenticated = process.env.AUTHENTICATED === 'false'; // Check if unauthenticated mode is enabled
 
-				for ( const paramName of presignedParamNames ) {
-					if ( args[paramName] ) {
-						presignedParams[paramName] = args[paramName];
-					}
-				}
+	// Create the S3 client configuration
+    const s3Config: S3ClientConfig = {
+        ...config,
+	 	forcePathStyle: true, // Required for MinIO
+	};
 
-				const headers: typeof request.headers = {};
-				for ( const header in request.headers ) {
-					if ( signedHeaders.includes( header.toLowerCase() ) ) {
-						headers[header] = request.headers[header];
-					}
-				}
-				request.query = presignedParams;
+    // Remove credentials for unauthenticated requests
+    if (unauthenticated) {
+        s3Config.credentials = {
+			accessKeyId: "",
+			secretAccessKey: "",
+		}
+    }
```

Comment:
The formatting for all the changes looks off and needs to be fixed, seems like you may be using spaces for indentation instead of tabs.